core/src/appleMain/kotlin/Peripheral.kt

```diff
@@ -224,7 +225,12 @@ public class ApplePeripheral internal constructor(
     }
 
     public override suspend fun connect() {
-        connectJob.updateAndGet { it ?: connectAsync() }!!.await()
+        // Check CBCentral State since connecting can result in an api misuse message
+        if (centralManager.delegate.state.value == CBCentralManagerStatePoweredOn) {
+            connectJob.updateAndGet { it ?: connectAsync() }!!.await()
+        } else {
+            throw NotReadyException("Attempted to connect to device before Bluetooth is powered on")
+        }
```

Comment:
To mirror the implementation on Android, let's add the following function to this class:

```kotlin
private fun checkBluetoothState(expected: CBCentralManagerState) {
    fun nameFor(value: Int) = when (value) {
        CBCentralManagerStatePoweredOff -> "PoweredOff"
        CBCentralManagerStatePoweredOn -> "PoweredOn"
        CBCentralManagerStateResetting -> "Resetting"
        CBCentralManagerStateUnauthorized -> "Unauthorized"
        CBCentralManagerStateUnknown -> "Unknown"
        CBCentralManagerStateUnsupported -> "Unsupported"
        else -> "Unknown"
    }
    val actual = centralManager.delegate.state.value
    if (expected != actual) {
        val actualName = nameFor(actual)
        val expectedName = nameFor(expected)
        throw BluetoothDisabledException("Bluetooth state is $actualName ($actual), but $expectedName ($expected) was required.")
    }
}
```

Then this code would change to:

```suggestion
        // Check CBCentral State since connecting can result in an api misuse message
        checkBluetoothState(CBCentralManagerStatePoweredOn)
        connectJob.updateAndGet { it ?: connectAsync() }!!.await()
```

_Note that this does throw `BluetoothDisabledException` if state is not powered on, which is probably poorly named, but is consistent with Android implementation — `BluetoothDisabledException` should probably be renamed in the future, or a new (more appropriately named exception) should be added; but this can be saved for a future/separate PR._